onnxruntime/core/session/ep_plugin_provider_interfaces.cc

```diff
@@ -0,0 +1,270 @@
+    ort_ep_->ReleaseNodeComputeInfos(ort_ep_.get(), api_node_compute_infos_.data(),
+                                     api_node_compute_infos_.size());
+  }
+}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+PluginExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_viewer,
+                                       const IKernelLookup& kernel_lookup,
+                                       const GraphOptimizerRegistry& graph_optimizer_registry,
+                                       IResourceAccountant* resource_accountant) const {
+  ORT_UNUSED_PARAMETER(graph_optimizer_registry);  // TODO: Add support
+  ORT_UNUSED_PARAMETER(resource_accountant);       // TODO: Add support? Not used by prioritized EPs
+  ORT_UNUSED_PARAMETER(kernel_lookup);             // TODO: Add support? Not used by prioritized EPs, so probably not needed?
+
+  std::unique_ptr<EpGraph> ep_graph = nullptr;
+  if (Status status = EpGraph::Create(graph_viewer, /*parent_node*/ nullptr, ep_graph); !status.IsOK()) {
+    LOGS_DEFAULT(ERROR) << "Failed to create OrtGraph: " << status.ToString();
+    return {};
+  }
+
+  OrtEpGraphSupportInfo api_graph_support_info(*ep_graph);
+  UniqueOrtStatus status(ort_ep_->GetCapability(ort_ep_.get(), ep_graph->ToExternal(), &api_graph_support_info),
+                         OrtApis::ReleaseStatus);
+
+  if (status != nullptr) {
+    LOGS_DEFAULT(ERROR) << "OrtEp::GetCapability() failed with error: " << ToStatus(status.get()).ToString();
+    return {};
+  }
+
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+  result.reserve(api_graph_support_info.node_groupings.size());
+  if (api_graph_support_info.node_groupings.empty()) {
+    return {};
+  }
+
+  ModelMetadefIdGenerator generator;
+
+  // Create ComputeCapability instances from OrtEpGraphSupportInfo::NodeGrouping instances.
+  for (const OrtEpGraphSupportInfo::NodeGrouping& node_grouping : api_graph_support_info.node_groupings) {
+    if (node_grouping.kind == OrtEpGraphSupportInfo::NodeGroupingKind::kSingleAssignedNode) {
+      auto indexed_sub_graph = std::make_unique<IndexedSubGraph>();
+
+      indexed_sub_graph->nodes.push_back(node_grouping.nodes[0]->node.Index());
+      result.push_back(std::make_unique<ComputeCapability>(std::move(indexed_sub_graph)));
+    } else if (node_grouping.kind == OrtEpGraphSupportInfo::NodeGroupingKind::kFusedNode) {
+      std::unordered_set<const Node*> node_set;
+      node_set.reserve(node_grouping.nodes.size());
+      for (const EpNode* ep_node : node_grouping.nodes) {
+        node_set.insert(&ep_node->node);
+      }
+
+      // We now require the OrtEp to only provide individual groups of supported nodes that each maps to exactly
+      // one ComputeCapability. Calling utils::CreateSupportedPartitions() may create multiple ComputeCapability
+      // instances, and if so, log an error and return.
+      std::vector<std::unique_ptr<ComputeCapability>> capabilities = utils::CreateSupportedPartitions(
```

Comment:
Instead of calling CreateSupportedPartitions here, could we instead provide that more directly to the EP implementation via OrtEpApi? EP provides a function pointer for IsSupported and OnGroupClosed.